src/coreclr/jit/importer.cpp

```diff
@@ -14411,15 +14411,27 @@ void Compiler::impImportBlockCode(BasicBlock* block)
                     impStackTop(0);
                 }
 
-                // If the expression to dup is simple, just clone it.
-                // Otherwise spill it to a temp, and reload the temp
-                // twice.
                 StackEntry se   = impPopStack();
                 GenTree*   tree = se.val;
                 tiRetVal        = se.seTypeInfo;
                 op1             = tree;
 
-                if (!opts.compDbgCode && !op1->IsIntegralConst(0) && !op1->IsFPZero() && !op1->IsLocal())
+                // If the expression to dup is simple, just clone it.
+                // Otherwise spill it to a temp, and reload the temp twice.
+                bool needsTemp = true;
+                if (opts.OptimizationEnabled())
+                {
+                    if (op1->IsIntegralConst(0) || op1->IsFPZero() || op1->IsLocal())
```

Comment:
I know it was already the case, but `IsFPZero` here is likely a bug.

It's going to catch both `+0.0` and `-0.0` and only the first is "simple". We should be using `IsFloatPositiveZero` instead which ensures its only `+0.0`.

Reply:
Thanks! addressed